### PR TITLE
Native lookup fixes

### PR DESCRIPTION
**Pull request by chagai95**
## Type of change

- [ ] Feature
- [X] Bugfix
- [ ] Technical
- [ ] Other :

## Content
refactor - better naming, return native user id and not sip user id and create a dm with the native user instead of with the sip user

## Motivation and context
When dialing an extension which has a native user behind it, a native matrix call happens instead of a matrix sip call because the user id of the sip user was used instead of the native one - so the sip virtual lookup (which comes in this case after the native lookup) failed. Otherwise also creating a native room when one does not exist seems like the right thing to do and the code looks more clear with these names.

## Tests

- private setup - you should have access in Aarenet/Element room

## Tested devices

- [X] Physical
- [ ] Emulator
- OS version(s):

## Checklist

- [X] Changes has been tested on an Android device or Android emulator with API 21
- [ ] UI change has been tested on both light and dark themes
- [ ] Accessibility has been taken into account. See https://github.com/vector-im/element-android/blob/develop/CONTRIBUTING.md#accessibility
- [X] Pull request is based on the develop branch
- [x] Pull request includes a new file under ./changelog.d. See https://github.com/vector-im/element-android/blob/develop/CONTRIBUTING.md#changelog
- [ ] Pull request includes screenshots or videos if containing UI changes
- [X] Pull request includes a [sign off](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#sign-off)
- [X] You've made a self review of your PR
- [ ] If you have modified the screen flow, or added new screens to the application, you have updated the test [UiAllScreensSanityTest.allScreensTest()](https://github.com/vector-im/element-android/blob/main/vector/src/androidTest/java/im/vector/app/ui/UiAllScreensSanityTest.kt#L73)

Signed-off-by: Chagai Friedlander @me:chagai.website

### CODE REVIEW

**Review by bmarty — COMMENTED** [review]
Thanks, for the PR.
LGTM, but I will defer to @ganfra and/or @dbkr for the logic.

**Comment by bmarty** [comment]
(can you check the compilation issues please?)

**Comment by chagai95** [comment]
> (can you check the compilation issues please?)

Couldn't figure out where the actual errors are... can you help me with that?

File: vector/src/main/java/im/vector/app/features/call/dialpad/DialPadLookup.kt [hunk 1]
```diff
@@ -42,18 +42,23 @@ class DialPadLookup @Inject constructor(
...
             if (nativeUserId == session.myUserId) {
                 throw Failure.NumberIsYours
             }
-            session.roomService().getExistingDirectRoomWithUser(nativeUserId)
-            // if there is not, just create a DM with the sip user
-                    ?: directRoomHelper.ensureDMExists(sipUserId)
-        } else {
-            // do the same if there is no corresponding native user.
-            directRoomHelper.ensureDMExists(sipUserId)
+            var nativeRoomId = session.getExistingDirectRoomWithUser(nativeUserId)
```
Comment (bmarty) [inline vector/src/main/java/im/vector/app/features/call/dialpad/DialPadLookup.kt:50]: You are maybe not using the latest develop from mainline?
The code should be session **.roomService()** .getExistingDirectRoomWithUser(nativeUserId) now

File: vector/src/main/java/im/vector/app/features/call/dialpad/DialPadLookup.kt [hunk 2]
```diff
@@ -42,18 +42,23 @@ class DialPadLookup @Inject constructor(
...
+            if (nativeRoomId == null) {
+                // if there is no existing native room with the existing native user,
+                // just create a DM with the native user
+                nativeRoomId = directRoomHelper.ensureDMExists(nativeUserId)
+            }
+            Timber.d("lookupPhoneNumber with nativeUserId: $nativeUserId and nativeRoomId: $nativeRoomId")
```
Comment (bmarty) [inline vector/src/main/java/im/vector/app/features/call/dialpad/DialPadLookup.kt:56]: An import of `Timber` is missing.

File: vector/src/main/java/im/vector/app/features/call/dialpad/DialPadLookup.kt (hunk 1 above)
Comment (chagai95) [reply vector/src/main/java/im/vector/app/features/call/dialpad/DialPadLookup.kt:50]: oops, completely missed that - copy pasted that part of the change from our fork, sorry.

File: vector/src/main/java/im/vector/app/features/call/dialpad/DialPadLookup.kt (hunk 2 above)
Comment (chagai95) [reply vector/src/main/java/im/vector/app/features/call/dialpad/DialPadLookup.kt:56]: Yep, sorry 🙈

**Comment by bmarty** [comment]
(it's confirmed I am not the only one to create PR without even having compiled the code locally!)

**Comment by chagai95** [comment]
> > > (can you check the compilation issues please?)
> > 
> > 
> > Couldn't figure out where the actual errors are... can you help me with that?
> 
> I have added some comments and the error details can be found on clicking on "Details" (in case you are not aware of that):
> 
> image

I tried that, but still couldn't really figure out where the errors are... sorry... also sorry for not compiling locally 🙈
Here is a screen cap of me trying to understand the failing builds:
https://aarenet-my.sharepoint.com/:v:/p/chagai_friedlander/EcN-uVEqkIxGkfvg3lPYKVQBGVsf0T_6BMwpUkVH7VGilA

**Review by dbkr — APPROVED** [review]
The (pre-existing) variable names in here are quite confusing, but the logic of the change looks correct.

**Comment by chagai95** [comment]
I think everything passed?

```
* What went wrong:
Execution failed for task ':sonarqube'.
> Project not found. Please check the 'sonar.projectKey' and 'sonar.organization' properties, the 'SONAR_TOKEN' environment variable, or contact the project administrator
```

**Review by bmarty — APPROVED** [review]
The error reported by the CI can be ignored; we will try to fix that.
Thanks for the update, and thanks @dbkr for the review!

**Pull request merged**